packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/dropNode/useDropNodeStyle.ts

```diff
@@ -13,19 +13,22 @@
 
 import { useTheme } from '@material-ui/core/styles';
 import { useContext } from 'react';
+import { DiagramContext } from '../../contexts/DiagramContext';
+import { DiagramContextValue } from '../../contexts/DiagramContext.types';
 import { DropNodeContext } from './DropNodeContext';
 import { DropNodeContextValue } from './DropNodeContext.types';
 import { useDropNodeStyleValue } from './useDropNodeStyle.types';
 
 export const useDropNodeStyle = (nodeId: string): useDropNodeStyleValue => {
+  const { readOnly } = useContext<DiagramContextValue>(DiagramContext);
   const { draggedNode, targetNodeId, compatibleNodeIds } = useContext<DropNodeContextValue>(DropNodeContext);
   const theme = useTheme();
 
   const isCompatibleDropTarget: boolean = compatibleNodeIds.includes(nodeId);
   const isSelectedDropTarget: boolean = nodeId === targetNodeId;
   const style: React.CSSProperties = {};
 
-  if (draggedNode !== null) {
+  if (draggedNode !== null && !readOnly) {
```

Comment:
Can we even have a `draggedNode` in `readOnly`? It shouldn't be possible to move anything at all, no?